src/rgw/rgw_sal_motr.cc

```diff
@@ -2421,7 +2429,16 @@ int MotrAtomicWriter::write()
     if (left < bs)
       bs = obj.get_optimal_bs(left);
     if (left < bs) {
-      acc_data.append_zero(bs - left);
+      // Pad to allign with unit size (and not the group size) and 
+      // set M0_ENF_NO_RMW flag to force no RMW
+      obj.mobj->ob_entity.en_flags |= M0_ENF_NO_RMW;
+      uint64_t lid = M0_OBJ_LAYOUT_ID(obj.meta.layout_id);
+      unsigned unit_sz = m0_obj_layout_id_to_unit_size(lid);
+
+      unsigned left_alligned_unit_sz = roundup(left, unit_sz);
+      ldpp_dout(dpp, 20) <<__func__<< " Padding [" << (left_alligned_unit_sz - left) << "] bytes" << dendl;
+      acc_data.append_zero(left_alligned_unit_sz - left);
```

Comment:
Do we still need padding ?

Reply:
yes, padding is needed to the last IO to make it aligned to the unit size boundary. E.g., if layout id of object is 0xb (i.e. 11), with unit size = 4MB, if the last IO is 3.5MB, then we have to pad it 0.5MB (to make it 4MB aligned)